truss/templates/triton/root/generate_config.py

```diff
@@ -0,0 +1,196 @@
+    if not hasattr(pydantic_field_annotation, "__origin__"):
+        return False
+    return pydantic_field_annotation.__origin__ == list
+
+
+def get_triton_type(pydantic_field: FieldAnnotationType) -> str:
+    if is_conlist(pydantic_field):
+        # If the type is a list, get the type of the list elements
+        pydantic_field = pydantic_field.__args__[0]  # type: ignore
+
+    type_name = (
+        pydantic_field.__name__
+        if hasattr(pydantic_field, "__name__")
+        else str(pydantic_field)
+    )
+
+    if type_name not in TYPE_MAPPING:
+        raise TypeError(f"Unsupported type: {type_name}")
+
+    return TYPE_MAPPING[type_name]
+
+
+def get_triton_dims(pydantic_type: Type) -> List[int]:
+    if is_conlist(pydantic_type.annotation):
+        dims = next(
+            (m for m in pydantic_type.metadata if hasattr(m, "min_length")), None
+        )
+        if dims and dims.min_length == dims.max_length:
+            return [-1, dims.min_length]
+        else:
+            return [-1, -1]
+    return [-1]
+
+
+def validate_user_model_class(module: ModuleType, required_attributes: List[str]):
+    """
+    Validates that a user model class has the required attributes.
+
+    Args:
+        module (ModuleType): The module containing the user model class.
+        required_attributes (list): A list of required attributes.
+
+    Raises:
+        AttributeError: If a required attribute is missing.
+    """
+    for attribute in required_attributes:
+        if not hasattr(module, attribute):
+            raise AttributeError(
+                f"Truss model class is missing {attribute} attribute. For Triton, \
+                we require a Pydantic model class that corresponds to your model \
+                input and model output."
+            )
+
+
+def generate_config_pbtxt(
+    input_class: Type[BaseModel],
+    output_class: Type[BaseModel],
+    template_path: Path,
+    template_name: str = "config.pbtxt.jinja",
+    max_batch_size: int = 1,
+    num_replicas: int = 1,
+    dynamic_batch_delay_ms: int = 0,
+    is_gpu: bool = False,
+) -> str:
+    config_params = {
+        "max_batch_size": max_batch_size,
+        "num_replicas": num_replicas,
+        "is_gpu": is_gpu,
+    }
+    template = read_template_from_fs(template_path, template_name)
+    input_cls_field_descriptors = FieldDescriptor.from_pydantic_class(input_class)
+    output_cls_field_descriptors = FieldDescriptor.from_pydantic_class(output_class)
+    inputs = [field.to_dict() for field in input_cls_field_descriptors]
+    outputs = [field.to_dict() for field in output_cls_field_descriptors]
+    if dynamic_batch_delay_ms > 0:
+        config_params["dynamic_batching_delay_microseconds"] = dynamic_batch_delay_ms
+    return template.render(inputs=inputs, outputs=outputs, **config_params)
+
+
+def main():
+    model_repository_path = Path("model")
+    user_truss_path = model_repository_path / "1" / "truss"
```

Comment:
there's a lot going on in this `main` function, I think it might be more readable if you broke this into a few different functions